# CAPSTONE PROJECT

Name : RIFANA RAJA

Batch : DA|ML-DAF50

Submission Date : 29-09-2026

URL of the Dataset

https://www.kaggle.com/datasets/sgseba/bike-data-sales-2020-2024-synthetic

# End-to-End Sales Analysis

## **Global Performance and Customer Demographics Optimization**

**Introduction**

This project, titled **Global Performance and Customer Demographics Optimization**, develops an End-to-End Sales Analytics pipeline using Python to transform raw retail transaction data into actionable insights. The temporal range of this project covers five years, from January 2020 to December 2024, integrating operational attributes such as order numbers and transaction dates with customer demographics including age and gender, as well as geographic dimensions like country and state. By combining these elements, the system enables a comprehensive analysis of sales performance, buyer behavior, and regional trends, supporting strategic decision-making for business scaling and customer engagement optimization.








## **Problem Statement**

In the modern retail ecosystem, companies face significant challenges in optimizing their omnichannel sales and understanding diverse customer demographics. Traditional spreadsheet-driven methods are manual, error-prone, and inefficient for parsing large multi-year transactional datasets. This creates data silos and delays in identifying crucial business trends.

Without an automated analytical pipeline, the business struggles with three critical gaps:
1. **Dirty Data & Typos:** Inconsistent data entries (such as regional variations and gender typos) lead to inaccurate reporting and flawed segmentations.
2. **Channel Friction:** A lack of clarity on how customer purchasing behavior shifts between Online platforms and Physical Stores over time and across different days (Weekdays vs. Weekends).
3. **Demographic Gaps:** Inability to pinpoint which specific age cohorts and geographic regions (like individual countries) drive the maximum profit and transactional volume.

### **The Solution**
This Capstone Project solves these problems by engineering an automated, code-driven Data Analytics pipeline using **Python and Pandas**. By systematically cleaning structural data anomalies, creating enriched features (`Is_Weekend`, `Order_Type`), and generating high-impact **Plotly visualizations**, this pipeline transforms raw retail transactions into clear, actionable business strategies to optimize market reach and boost omnichannel efficiency

Key Objectives





*   Automate Data Cleaning: Fix text errors and typos automatically using Python and Pandas instead of correcting them manually.

*   Track Market Performance: Monitor and compare sales performance across different countries.

*   Profile Demographics: Categorize transaction volumes based on customer gender and different age groups.


*   Build a Code-Driven Pipeline: Replace traditional spreadsheet tools entirely with automated Python data workflows.



**PHASE - 1**

# ***Data Loading***

For data preprocessing, I  used Python and Pandas instead of manual spreadsheet work or other options. Pandas is chosen because it handles tabular data faster and more efficiently than standard libraries. It provides built-in tools like pd.to_datetime() and dropna()  etc.., that let us easily clean missing data, fix formatting errors, and handle messy text anomalies in just a few lines of code.

In [91]:
import pandas as pd

In [92]:

df=pd.read_excel('bike_sales_2020_2024-selected-columns.csv.xlsx')
df


,Sales_Order #,Date,Day,Month,Year,Customer_Age,Age_Group,Customer_Gender,Country,State
0,ON-000261695,2020-01-01,1.0,January,2020,28,Youth (<25),F,United States,Queensland
1,ST-000261696,2020-01-01,1.0,January,2020,40,Adults (35-64),M,United States,Seine et Marne
2,ON-000261697,2020-01-02,2.0,January,2020,65,Young Adults (25-34),M,Australia,Seine Saint Denis
3,ON-000261698,2020-01-02,2.0,January,2020,24,Young Adults (25-34),M,United States,California
4,ON-000261699,2020-01-03,3.0,January,2020,40,NaN,F,Germany,Nordrhein-Westfalen
...,...,...,...,...,...,...,...,...,...,...
5258,ST-000266953,2024-12-29,29.0,December,2024,61,Youth (<25),M,United States,Seine Saint Denis
5259,ST-000266954,2024-12-30,30.0,December,2024,36,Young Adults (25-34),M,United States,Nordrhein-Westfalen
5260,ON-000266955,2024-12-30,30.0,December,2024,25,Young Adults (25-34),F,Germany,Nordrhein-Westfalen
5261,ST-000266956,2024-12-31,31.0,December,2024,62,Young Adults (25-34),M,Australia,South Australia



# ***Initial Overview of the Dataset***

In [93]:
# Shape of the dataset

df.shape

(5263, 10)

The dataset contain 5263 rows and 10 columns

In [94]:
# Datatypes of each columns

df.dtypes

,0
Sales_Order #,object
Date,datetime64[ns]
Day,float64
Month,object
Year,int64
Customer_Age,int64
Age_Group,object
Customer_Gender,object
Country,object
State,object


This dataset contains 6 object type columns, 3 Numerical columns and 1 Date colum

In [95]:
# count the missing values in each column of the dataset

df.isnull().sum()

,0
Sales_Order #,0
Date,0
Day,66
Month,0
Year,0
Customer_Age,0
Age_Group,67
Customer_Gender,0
Country,0
State,0


The dataset contains 66 missing values in the Day column and 67 missing values in the Age_Group column, indicating incomplete records for those fields.

# ***Initial Observations***

df.head() is used to preview the first rows and check dataset structure.

In [96]:
# head of the dataset

df.head()

,Sales_Order #,Date,Day,Month,Year,Customer_Age,Age_Group,Customer_Gender,Country,State
0,ON-000261695,2020-01-01,1.0,January,2020,28,Youth (<25),F,United States,Queensland
1,ST-000261696,2020-01-01,1.0,January,2020,40,Adults (35-64),M,United States,Seine et Marne
2,ON-000261697,2020-01-02,2.0,January,2020,65,Young Adults (25-34),M,Australia,Seine Saint Denis
3,ON-000261698,2020-01-02,2.0,January,2020,24,Young Adults (25-34),M,United States,California
4,ON-000261699,2020-01-03,3.0,January,2020,40,NaN,F,Germany,Nordrhein-Westfalen


df.tail() is used to preview the last rows of the dataset and check how the data ends.

In [97]:
# Tail of the dataset

df.tail()

,Sales_Order #,Date,Day,Month,Year,Customer_Age,Age_Group,Customer_Gender,Country,State
5258,ST-000266953,2024-12-29,29.0,December,2024,61,Youth (<25),M,United States,Seine Saint Denis
5259,ST-000266954,2024-12-30,30.0,December,2024,36,Young Adults (25-34),M,United States,Nordrhein-Westfalen
5260,ON-000266955,2024-12-30,30.0,December,2024,25,Young Adults (25-34),F,Germany,Nordrhein-Westfalen
5261,ST-000266956,2024-12-31,31.0,December,2024,62,Young Adults (25-34),M,Australia,South Australia
5262,ST-000266957,2024-12-31,31.0,December,2024,68,Adults (35-64),F,United Kingdom,Nordrhein-Westfalen


df.describe() is used to generates summary statistics of numerical columns, showing count, mean, standard deviation, min, max, and quartiles to understand data distribution.

---



In [98]:
# summary status of numerical columns

df.describe()

,Date,Day,Year,Customer_Age
count,5263,5197.000000,5263.000000,5263.000000
mean,2022-07-03 01:13:03.203495936,15.751395,2022.000000,43.325290
min,2020-01-01 00:00:00,1.000000,2020.000000,18.000000
25%,2021-04-14 00:00:00,8.000000,2021.000000,30.000000
50%,2022-06-28 00:00:00,16.000000,2022.000000,43.000000
75%,2023-09-19 00:00:00,23.000000,2023.000000,56.000000
max,2024-12-31 00:00:00,31.000000,2024.000000,69.000000
std,NaN,8.794143,1.407344,15.036173


df.info()is used for gives a concise summary of the dataset, showing column names, non‑null counts, data types, and memory usage.

In [99]:
# Structure of the dataset

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5263 entries, 0 to 5262
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   Sales_Order #    5263 non-null   object        
 1   Date             5263 non-null   datetime64[ns]
 2   Day              5197 non-null   float64       
 3   Month            5263 non-null   object        
 4   Year             5263 non-null   int64         
 5   Customer_Age     5263 non-null   int64         
 6   Age_Group        5196 non-null   object        
 7   Customer_Gender  5263 non-null   object        
 8   Country          5263 non-null   object        
 9   State            5263 non-null   object        
dtypes: datetime64[ns](1), float64(1), int64(2), object(6)
memory usage: 411.3+ KB


df.nunique() is used to get the count of unique values in each column, helping identify distinct categories and spot inconsistencies.

In [100]:
# Check how  distinct values exist in the dataset

df.nunique()

,0
Sales_Order #,5263
Date,1827
Day,31
Month,19
Year,5
Customer_Age,52
Age_Group,3
Customer_Gender,7
Country,17
State,23


df.duplicated().sum() used to get  the total number of duplicate rows in the dataset, helping check data quality

In [101]:
# find duplicate values in the dataset

df.duplicated().sum()

np.int64(0)

*df*.columns is used to get the lists all column names in the dataset, helping check structure and available attributes.

In [102]:
# Columns name

df.columns

Index(['Sales_Order #', 'Date', 'Day', 'Month', 'Year', 'Customer_Age',
       'Age_Group', 'Customer_Gender', 'Country', 'State'],
      dtype='object')

 **PHASE - 2**

# ***Data Pre-processing Workflow***

I preprocessed  this sales dataset by handling missing entries and standardizing category typos across gender and country fields. Using Pandas, I  remove duplicates, correct date formats, and create new derived columns like 'Order Type'  and 'Is_Weekend' to prepare the structured data for clean reporting and aggregation.

Step - 1

***Handling Missing Values***

count the missing values in each column of the dataset



In [103]:
df.isnull().sum()

,0
Sales_Order #,0
Date,0
Day,66
Month,0
Year,0
Customer_Age,0
Age_Group,67
Customer_Gender,0
Country,0
State,0


The dataset contains 66 missing values in the Day column and 67 missing values in the Age_Group column, indicating incomplete records for those fields.

step -2

***Remove Whitespaces & Standardize the text***

1-Remove accidental leading/trailing whitespaces from column names





In [104]:

df.columns = df.columns.str.strip()
df.columns


Index(['Sales_Order #', 'Date', 'Day', 'Month', 'Year', 'Customer_Age',
       'Age_Group', 'Customer_Gender', 'Country', 'State'],
      dtype='object')

 After running The .str.strip() function targets every single column name and automatically slices off any accidental spaces hiding at the front or back.

Before running: The Customer_Gender column contained typos and inconsistent values like "Feamle", "Fmale", "Mael", which could cause errors in grouping or analysis.



In [105]:

df['Customer_Gender'] = df['Customer_Gender'].str.strip()

df['Customer_Gender'] = df['Customer_Gender'].replace({
    'Feamle':'F','Fmale':'F','Femal':'F','F':'F',
    'Male':'M','Mael':'M'
})

print(df.iloc[12:18]['Customer_Gender'])


12    F
13    M
14    M
15    F
16    F
17    F
Name: Customer_Gender, dtype: object


After running: All gender values are standardized into  "F" and "M".

Before running: The Country column contained typos and variations like "US", "United States", "Germnay", "Austraila", "United Kindom", which made analysis inconsistent.

In [106]:

df['Country'] = df['Country'].replace({
    'US':'United States','United  States':'United States','United States ':'United States',
    'Germnay':'Germany','Deutschland':'Germany',
    'Austraila':'Australia','Aussie':'Australia',
    'United Kindom':'United Kingdom','UK':'United Kingdom','U.K.':'United Kingdom'
})

print(df.iloc[20:25]['Country'])


20    United States
21           France
22           France
23           France
24           France
Name: Country, dtype: object


After running: All these variations were standardized into clean names like "United States", "Germany", "Australia", and "United Kingdom", reducing duplicates and ensuring consistent country categories.

Text columns like Country and Customer_Gender contained accidental padding spaces and multiple spelling variations such as " United States", "Feamle", and "Fmale", which caused the dataset to falsely report 17 unique countries and 7 unique genders due to inconsistent entries.

In [107]:

string_cols = df.select_dtypes(include=['object']).columns
for col in string_cols:
    df[col] = df[col].astype(str).str.strip()

print("Unique Countries:", df['Country'].unique())
print("Unique Genders:", df['Customer_Gender'].unique())


Unique Countries: ['United States' 'Australia' 'Germany' 'United Kingdom' 'France' 'Canada']
Unique Genders: ['F' 'M']


After running: This process automatically converts all object columns to clean strings, strips off front and back spaces, and applies explicit category mappings, standardizing values so the data consolidates into 6 unique countries and 2 unique genders for flawless aggregation.

Step - 3

***Correct Data Types***

Before running: The Date column appeared as datetime but could still contain inconsistent or invalid entries like '2020-13-01'

In [108]:

df['Date'] = pd.to_datetime(df['Date'], errors='coerce')
df = df.dropna(subset=['Date'])

print(df.loc[5, 'Date'])
print(df.shape)


2020-01-03 00:00:00
(5263, 10)


After running:
 The column is fully standardized to datetime64[ns]. As a defensive step, any invalid dates would have been converted to NaT and dropped using .dropna() — in this dataset, df.shape stayed at (5263, 10) before and after this step, confirming no invalid dates actually existed and the Date column was already clean.

Step - 4

***Create Derived Columns***

Before running: The dataset only had the full Date column. The separate Day, Month, and Year columns were either missing or contained incomplete values.

In [109]:
df['Day'] = df['Date'].dt.day.astype(int)

df['Month'] = df['Date'].dt.strftime('%B')

df['Year'] = df['Date'].dt.year.astype(int)

print(df.loc[5, 'Date'])
print(df.loc[5, ['Day','Month','Year']])
print(df.isnull().sum())


2020-01-03 00:00:00
Day            3
Month    January
Year        2020
Name: 5, dtype: object
Sales_Order #      0
Date               0
Day                0
Month              0
Year               0
Customer_Age       0
Age_Group          0
Customer_Gender    0
Country            0
State              0
dtype: int64








After running: The Day, Month, and Year columns are successfully extracted from the clean Date column. Since the missing dates are fixed first, the values safely convert into whole numbers (integers) without any code errors. This gives us clean time filters for accurate business reporting.

Before running:  
  The dataset had raw ages but the Age_Group column was incomplete and miscategorized. The goal was to rebuild it accurately using a custom Python function.

In [110]:

df['Customer_Age'] = df['Customer_Age'].astype(int)

def categorized_age(age):
    if age < 25:
        return 'Youth (<25)'
    elif 25 <= age <= 34:
        return 'Young Adults (25-34)'
    elif 35 <= age <= 64:
        return 'Adults (35-64)'
    else:
        return 'Seniors (65+)'


df['Age_Group'] = df['Customer_Age'].apply(categorized_age)

print(df['Age_Group'].isnull().sum())
print(df[['Customer_Age', 'Age_Group']].head(10))


0
   Customer_Age             Age_Group
0            28  Young Adults (25-34)
1            40        Adults (35-64)
2            65         Seniors (65+)
3            24           Youth (<25)
4            40        Adults (35-64)
5            50        Adults (35-64)
6            23           Youth (<25)
7            44        Adults (35-64)
8            37        Adults (35-64)
9            51        Adults (35-64)


After running :
The function fully corrected the Age_Group column, removed blanks and errors, and ensured clean, reliable categories for demographic analysis.

Before running:  
The dataset only had raw dates in the Date column, so it wasn’t possible to directly identify weekend purchases.

In [111]:

df["Is_Weekend"] = df["Date"].dt.dayofweek.isin([5, 6]).astype(int)

print(df[['Date','Is_Weekend']].head(10))


        Date  Is_Weekend
0 2020-01-01           0
1 2020-01-01           0
2 2020-01-02           0
3 2020-01-02           0
4 2020-01-03           0
5 2020-01-03           0
6 2020-01-04           1
7 2020-01-04           1
8 2020-01-05           1
9 2020-01-05           1


After running:  
A new Is_Weekend column was created, marking weekend transactions with 1 and weekday transactions with 0, making it easy to analyze weekend buying behavior.

Before running:  
The dataset only had raw values in the Sales_Order # column, such as ON12345 or ST67890, which did not show whether the order was online or store‑based.

In [112]:

def order_type(order_id):

    if str(order_id).startswith('ON'):
        return 'Online'
    else:
        return 'Store'

df['Order_Type'] = df['Sales_Order #'].apply(order_type)

print(df[['Sales_Order #','Order_Type']].head(10))


  Sales_Order # Order_Type
0  ON-000261695     Online
1  ST-000261696      Store
2  ON-000261697     Online
3  ON-000261698     Online
4  ON-000261699     Online
5  ST-000261700      Store
6  ON-000261701     Online
7  ST-000261702      Store
8  ST-000261703      Store
9  ON-000261704     Online


After running:  
A new column Order_Type was created, classifying each order as “Online” if the ID starts with ON and “Store” otherwise, making the origin of orders clear and easy to analyze.

Step - 5

***Remove Duplicates***

In [113]:

duplicate_count=df.duplicated().sum()

print("Number of duplicate rows:", duplicate_count)


Number of duplicate rows: 0


The code checks for duplicate rows in the DataFrame. Since duplicate_count returned 0, it confirms that no duplicate records exist in the dataset.

Step - 6

***Aggregation/Summarization***

Before: The dataset only had individual order records, so there was no direct summary of how many orders each country had.

In [114]:

country_aggregation = df.groupby('Country').size().reset_index(name='Total_Orders')

print(country_aggregation)


          Country  Total_Orders
0       Australia           531
1          Canada           591
2          France           556
3         Germany           616
4  United Kingdom           567
5   United States          2402


After: By grouping with df.groupby('Country').size().reset_index(name='Total_Orders'), the new output table shows each country alongside its total number of orders, making the dataset easier to analyze at a country level.

Before: The dataset only had raw age group entries scattered across countries, so there was no clear summary of how many people fell into each category per country.

In [115]:

age_counts = df.groupby('Country')['Age_Group'].value_counts()
print(age_counts)


Country         Age_Group           
Australia       Adults (35-64)           322
                Young Adults (25-34)      95
                Youth (<25)               64
                Seniors (65+)             50
Canada          Adults (35-64)           324
                Young Adults (25-34)     123
                Youth (<25)               96
                Seniors (65+)             48
France          Adults (35-64)           311
                Young Adults (25-34)     111
                Youth (<25)               80
                Seniors (65+)             54
Germany         Adults (35-64)           363
                Young Adults (25-34)     123
                Youth (<25)               74
                Seniors (65+)             56
United Kingdom  Adults (35-64)           305
                Young Adults (25-34)     120
                Youth (<25)               74
                Seniors (65+)             68
United States   Adults (35-64)          1364
                Yo

After: By running df.groupby('Country')['Age_Group'].value_counts(), the output now shows a breakdown of counts for each age group within every country, making demographic analysis by country straightforward.

Before running: This code filters the dataset to include only online orders from the United States and resets the index so that the row numbers start fresh from 0, making the output easier to read.

In [116]:

us_online_sales = df[(df['Country'] == 'United States') &
                     (df['Order_Type'] == 'Online')].reset_index(drop=True)

print(us_online_sales[['Sales_Order #','Country','Order_Type',
                       'Customer_Age','Age_Group']].head())


  Sales_Order #        Country Order_Type  Customer_Age             Age_Group
0  ON-000261695  United States     Online            28  Young Adults (25-34)
1  ON-000261698  United States     Online            24           Youth (<25)
2  ON-000261701  United States     Online            23           Youth (<25)
3  ON-000261705  United States     Online            41        Adults (35-64)
4  ON-000261710  United States     Online            25  Young Adults (25-34)


After running: It shows the first few rows of the filtered data with clean sequential numbering (0, 1, 2, 3, 4) along with the order details, country, order type, customer age, and age group.

***Phase*** - **3**

# ***Exploratory Data Analysis (EDA)***

**EDA – Introduction**:
In this phase, I conducted a comprehensive exploratory analysis of the bike sales dataset from 2020 to 2024 to uncover customer behavior, hidden patterns, and market trends that shape overall performance.

The analysis includes:

*   Univariate profiling – examining age distributions and testing for normality, and identifying skewness or deviations.
*  Bivariate assessments – evaluating annual age stability and comparing age groups across countries.

*   Multivariate modeling – analyzing buying channel preferences and exploring correlations among key features.
*  Summary statistics – validating dataset consistency, diversity, and reliability for further visualization.

The central aim is to determine whether factors such as customer age, geography, sales channels, and timelines independently influence transaction volumes or share linear dependencies. This structured approach ensures that demographic stability, market dominance, and channel balance are clearly identified before moving into visualization.






In [117]:

import numpy as np
import plotly.express as px
import plotly.figure_factory as ff
import plotly.graph_objects as go
from plotly.subplots import make_subplots
from scipy import stats
from scipy.stats import skew, kurtosis




1.   **NumPy (numpy)**
NumPy is the fundamental library for scientific and mathematical computing in Python.
in this project, it was used to handle numerical arrays and perform smooth mathematical grid calculations required for density estimations (e.g., histograms, KDE ranges)

2.  **Plotly Figure Factory** (**plotly.figure_factory as ff**)This library is used to create specialized statistical charts that are not available in the standard Plotly Express package.
 It helped in building advanced visualizations like custom distplots and Q‑Q plots efficiently.

3. **Plotly Express** **(plotly.express as px)**
Plotly Express is a high‑level data visualization interface designed to generate sleek, interactive charts with minimal code.
 It was used to transform traditional static plots (histograms, box plots, scatter matrices, heatmaps, pie charts, line charts) into interactive elements.  


4. **Plotly Graph Objects**(**plotly.graph_objects as go**)
This module provides full control over the structural components of a Plotly figure.
 It allowed precise customization of individual traces, shapes, and complex plot layers (e.g., Q‑Q plot lines, KDE overlays, histogram bars).  

5.  **Plotly Subplots (plotly.subplots → make_subplots)**
This function is utilized to design multi‑plot layouts.
 It enabled rendering of multiple distinct charts (subplots) within a single canvas for efficient comparative analysis (e.g., Age vs Country + Channel Preference side by side).

6. **SciPy (scipy)**
SciPy provides advanced statistical and scientific tools.
 In this project, it was used for   probability plots, kernel density estimation (KDE), skewness, and kurtosis to analyze distribution properties.  



7.  **SciPy Stats (scipy.stats**)
A submodule of SciPy dedicated to statistical functions.
 It provided specific tools like skew, kurtosis, probplot, and gaussian_kde for deeper data analysis of customer age distributions.





***KPI Metrics***

Before:  
Here my aim to establish the overall business snapshot — total orders, average customer age, leading country, weekend activity, and channel split. These KPIs set the foundation for deeper analysis.

In [118]:

total_orders = df['Sales_Order #'].nunique()
avg_age = df['Customer_Age'].mean()
top_country = df['Country'].value_counts().idxmax()
weekend_orders = df['Is_Weekend'].mean() * 100

order_split = df['Order_Type'].value_counts(normalize=True) * 100
online_share = order_split.get('Online', 0)
store_share = order_split.get('Store', 0)

print("\n Total Orders:", total_orders)
print("\n Average Customer Age:", round(avg_age, 2))
print("\n Top Country:", top_country)
print("\n Weekend Orders %:", round(weekend_orders, 2))
print("\n Online Orders %:", round(online_share, 2))
print("\n Store Orders %:", round(store_share, 2))



 Total Orders: 5263

 Average Customer Age: 43.33

 Top Country: United States

 Weekend Orders %: 28.54

 Online Orders %: 51.05

 Store Orders %: 48.95


After:  
The dataset shows a healthy transaction volume. Adults around age 35–64 dominate purchases, the United States emerges as the top market, and weekend orders contribute a steady share. Online and store channels are balanced, confirming a stable integrated channel business model.

***Univariate Analysis***

**Orders by Age Group**

Before (Aim):  
The aim is to examine how orders are distributed across different age groups. This helps to identify which demographic segments contribute most to bike sales.



In [119]:

age_group_counts = df['Age_Group'].value_counts()
print("Orders by Age Group:")
print(age_group_counts)


Orders by Age Group:
Age_Group
Adults (35-64)          2989
Young Adults (25-34)    1057
Youth (<25)              714
Seniors (65+)            503
Name: count, dtype: int64


After (Finding):  
The results show that Adults (35–64) account for the largest share of orders, confirming that this age group functions as the primary economic driver. Young Adults and Seniors contribute steady portions, while Youth participation remains limited.

**Age Distribution**

Before (Aim):  
The aim is to analyze the distribution of customer age to check normality, skewness, and kurtosis. This helps in understanding whether the age variable is symmetrically spread or biased toward certain ranges.

In [120]:
 # Prepare age data, calculate skewness & kurtosis, and set up subplot layout

age_data = df["Customer_Age"].dropna()
age_skewness = skew(age_data)
age_kurtosis = kurtosis(age_data)

chart = make_subplots(
    rows=1, cols=2,
    subplot_titles=(
        "Q-Q Plot: Age vs Normal",
        f"Age Distribution<br>Skew={age_skewness:.2f} | Kurt={age_kurtosis:.2f}"
    )
)

# Create Q-Q plot comparing actual age distribution with theoretical normal distribution

(osm, osr), (slope, intercept, r) = stats.probplot(age_data, dist="norm")
chart.add_trace(go.Scatter(x=osm, y=osr, mode="markers", name="Data", marker=dict(color="blue")), row=1, col=1)
chart.add_trace(go.Scatter(x=osm, y=slope * osm + intercept, mode="lines", name="Normal Line", line=dict(color="red")), row=1, col=1)

# Build histogram of ages and overlay a scaled KDE curve for smooth distribution visualization

bin_counts, bin_edges = np.histogram(age_data, bins=20)
bin_width = bin_edges[1] - bin_edges[0]

x_vals = np.linspace(age_data.min(), age_data.max(), 200)
kde = stats.gaussian_kde(age_data)
kde_scaled = kde(x_vals) * len(age_data) * bin_width

chart.add_trace(go.Bar(x=bin_edges[:-1], y=bin_counts, name="Histogram", marker_color="blue", opacity=0.6), row=1, col=2)
chart.add_trace(go.Scatter(x=x_vals, y=kde_scaled, mode="lines", name="KDE", line=dict(color="red")), row=1, col=2)

chart.update_layout(height=600, width=1200, showlegend=False)
chart.show()


After (Finding):  
The skewness value (0.01) confirms the age distribution is essentially symmetric, not skewed toward either end. The kurtosis value (-1.22) indicates a platykurtic (flatter-than-normal) distribution — visible in the histogram as a fairly even spread of order counts across all age ranges rather than a sharp central peak. The Q-Q plot's S-curve shape, flattening at both the youngest and oldest ages, further reflects this short-tailed, evenly-spread distribution. Overall, customer age is broadly uniform across the 18–69 range rather than concentrated around a specific age band.

***Bivariate Analysis***

  **Average Age per Year**

Before (Aim):  
The aim is to calculate the average customer age for each year between 2020 and 2024. This helps to evaluate whether customer age remains stable over time or shows any significant shifts.

In [121]:

avg_age_by_year = df.groupby('Year')['Customer_Age'].mean().reset_index()
print("Average Customer Age by Year:")
print(avg_age_by_year)


Average Customer Age by Year:
   Year  Customer_Age
0  2020     43.323047
1  2021     43.334901
2  2022     43.244591
3  2023     43.059266
4  2024     43.673095


After (Finding):  
The results show that the average customer age remains consistent across the years, with only minor variations. This indicates that the age profile of customers is stable over time, supporting the observation of a balanced demographic trend.

**Customer Age by Year and Country × Age Group**

Before (Aim):  
The aim is to evaluate customer age stability across years and analyze how orders are distributed by country and age group. This helps to identify whether age trends remain consistent over time and which geographical regions dominate sales.

In [122]:

box = px.box(df, x='Year', y='Customer_Age', color='Year',
    color_discrete_sequence=px.colors.qualitative.Set2,
    title='<b>Customer Age Distribution Across Years</b>')

box.update_layout(xaxis_title='Year',yaxis_title='Customer Age',
    width=900,height=600,showlegend=False)

box.show()

# Pivot table as supporting calculation to summarize orders by Country and Age Group

pivot_table = pd.pivot_table(df,values='Sales_Order #',
    index='Country',columns='Age_Group',aggfunc='count')

print("\nOrders by Country and Age Group (Pivot Table):")
print(pivot_table)



Orders by Country and Age Group (Pivot Table):
Age_Group       Adults (35-64)  Seniors (65+)  Young Adults (25-34)  \
Country                                                               
Australia                  322             50                    95   
Canada                     324             48                   123   
France                     311             54                   111   
Germany                    363             56                   123   
United Kingdom             305             68                   120   
United States             1364            227                   485   

Age_Group       Youth (<25)  
Country                      
Australia                64  
Canada                   96  
France                   80  
Germany                  74  
United Kingdom           74  
United States           326  


After (Finding):  
The boxplot confirms that customer age distribution is stable across all years, with median values remaining close and no unusual outliers. This indicates that the customer base has remained consistent between 2020 and 2024. The pivot table highlights that the United States dominates across all age groups, with Adults (35–64) contributing the largest share of orders. Youth participation is comparatively lower, reinforcing the demographic stability observed earlier.

***Multivariate Analysis***

**Orders by Country × Age Group × Order Type**

Before (Aim):  
The aim is to calculate how orders are distributed across country, age group, and order type. This helps to identify demographic and geographical patterns in purchasing behavior and to see how online and store channels vary across different segments.

In [123]:

multi_orders = df.groupby(['Country','Age_Group','Order_Type'])['Sales_Order #'].count().reset_index()
print("Orders by Country, Age Group, and Order Type:")
print(multi_orders.head(10))


Orders by Country, Age Group, and Order Type:
     Country             Age_Group Order_Type  Sales_Order #
0  Australia        Adults (35-64)     Online            166
1  Australia        Adults (35-64)      Store            156
2  Australia         Seniors (65+)     Online             29
3  Australia         Seniors (65+)      Store             21
4  Australia  Young Adults (25-34)     Online             47
5  Australia  Young Adults (25-34)      Store             48
6  Australia           Youth (<25)     Online             29
7  Australia           Youth (<25)      Store             35
8     Canada        Adults (35-64)     Online            184
9     Canada        Adults (35-64)      Store            140


After (Finding):  
The results show that Adults (35–64) contribute the largest share of orders across all countries and order types. Both online and store channels are actively used, confirming that customer preferences are balanced across purchasing modes. The United States emerges as the leading market, with consistent dominance across all age groups.

**Correlation Heatmap**

Before (Aim):  
The aim is to examine correlations among features such as customer age, year, day, weekend activity, and order type. This helps to identify whether any strong linear relationships exist between variables that could influence customer behavior.

In [124]:

def check_online(order_id):
    return 1 if order_id == "Online" else 0

df_corr = df[["Customer_Age", "Day", "Year", "Is_Weekend"]].copy()
df_corr["Is_Online"] = df["Order_Type"].apply(check_online)

corr = df_corr.corr()

chart = px.imshow(
    corr,
    text_auto=".2f",
    color_continuous_scale="RdBu",
    zmin=-1,
    zmax=1,
    title="<b>Feature Correlation Matrix Heatmap</b>",
    labels=dict(color="Correlation")
)

chart.update_layout(width=800, height=600)
chart.update_xaxes(side="bottom")
chart.show()

After (Finding):

The correlation heatmap shows weak linear dependencies among the variables. Customer age, year, and weekend activity do not strongly influence each other, and order type remains independent of these features.

***Summary Statistics***

**Customer Age and Dataset Overview**

Before (Aim):  
The aim is to generate descriptive statistics for customer age and to check unique value counts across the dataset. This helps to validate dataset consistency, identify central tendencies, and confirm diversity across variables.

In [125]:

print("Summary Statistics for Customer Age:")
print(df['Customer_Age'].describe())

print("\nWeekend Orders %:", df['Is_Weekend'].mean() * 100)
print("\nUnique Value Counts:")
print(df.nunique())


Summary Statistics for Customer Age:
count    5263.000000
mean       43.325290
std        15.036173
min        18.000000
25%        30.000000
50%        43.000000
75%        56.000000
max        69.000000
Name: Customer_Age, dtype: float64

Weekend Orders %: 28.538856165684972

Unique Value Counts:
Sales_Order #      5263
Date               1827
Day                  31
Month                12
Year                  5
Customer_Age         52
Age_Group             4
Customer_Gender       2
Country               6
State                23
Is_Weekend            2
Order_Type            2
dtype: int64


After (Finding):  
The summary statistics confirm that customer age values are within a reasonable range, with a stable mean and median reflecting the dominance of adults. Weekend orders contribute a balanced share of transactions. The unique value counts highlight diversity across categorical variables, ensuring the dataset is reliable and well-structured for further visualization.

**Overall Performance and Findings**

The exploratory analysis of bike sales from 2020 to 2024 provides a clear picture of customer behavior, demographic stability, and market dominance.

**KPI Metrics**:
Transaction volumes remain healthy, with adults aged 35–64 driving the majority of sales. The United States consistently emerges as the top-performing country. Weekend orders contribute a steady share, and both online and store channels are balanced, confirming a stable integrated channel business model.

**Univariate Profiling**:  
The age distribution is essentially symmetric (skew ≈ 0.01) and platykurtic (kurtosis = -1.22), meaning customer ages are spread broadly and evenly across the 18–69 range rather than clustering around a specific age band. Adults remain the largest single segment by order count, but this reflects the width of that age bracket (35–64) rather than a true concentration in the underlying distribution

**Bivariate Assessments**:  
Average customer age remains consistent across years, with no major shifts in demographic trends. The boxplot confirms stability in age distribution from 2020 to 2024. The pivot table highlights that the United States dominates across all age groups, reinforcing its role as the strongest market cluster.

**Multivariate Modeling**:  
Grouped orders show that adults across countries actively use both online and store channels, reflecting balanced customer preferences. The correlation heatmap reveals weak linear dependencies among features such as age, year, and weekend activity, confirming that no single variable strongly influences others.

**Summary Statistics**:  
Descriptive statistics validate dataset consistency, with reasonable age ranges and balanced weekend activity. Unique value counts highlight diversity across variables, ensuring reliability for further visualization and reporting.

**Overall Performance**:  
The findings confirm a stable and balanced integrated channel business model, with adults aged 35–64 functioning as the primary economic driver. The United States stands out as the dominant geographical cluster, underscoring its critical role in overall performance. The dataset reflects demographic stability, balanced channel usage, and minimal feature dependencies, providing a strong foundation for visualization and strategic reporting in subsequent phases.

phase - 4

# Visualizations

In this phase, I aim to highlight the key characteristics of the dataset through visualization. By using charts and plots, I transform raw numbers into meaningful patterns that reveal customer behaviors, age distributions, regional differences, and sales trends. Visualization helps me not only to present the data clearly but also to uncover deeper insights

**Year-over-Year Dynamics of Online vs Store Orders**

Aim:  
The purpose of this visualization was to examine how the distribution of online and store orders changed across different years. By comparing both channels side by side, the goal was to identify shifts in customer purchasing behavior over time.

In [126]:
(1)
chart = px.histogram(
    df,
    x="Year",
    color="Order_Type",
    barmode="group",
    color_discrete_sequence=px.colors.qualitative.Set1,
    title="Year-over-Year Dynamics of Online vs Store Orders"
)

chart.update_traces(marker_line_color="black", marker_line_width=1)
chart.update_layout(xaxis_title="Year", yaxis_title="Count", title_x=0.5)
chart.show()


Findings:  
The grouped count plot revealed that 2022 showed the highest performance, with online orders rising above store orders. This indicates that customers strongly favored online purchasing in 2022, marking it as the peak year for digital sales growth.

**Monthly Sales Trend**

Aim:  
The objective of this visualization was to analyze how sales orders fluctuate across the months of the year. By arranging the months in chronological order, the goal was to identify seasonal patterns, peak periods, and any consistent dips in customer activity.

In [127]:
(2)
month_order = [
    "January","February","March","April","May","June",
    "July","August","September","October","November","December"
]

df["Month"] = pd.Categorical(df["Month"], categories=month_order, ordered=True)
monthly_sales = df.groupby("Month", observed=True)["Sales_Order #"].count().reset_index()

chart = px.line(
    monthly_sales,
    x="Month",
    y="Sales_Order #",
    markers=True,
    title="Monthly Sales Trend"
)

chart.update_traces(line_color="darkblue")
chart.update_layout(xaxis_title="Month", yaxis_title="Number of Orders")
chart.show()


Findings:  
The line plot revealed clear monthly variations. June showed the highest performance, marking it as the peak month for sales activity. After June, sales dropped sharply in July and gradually declined through the rest of the year, highlighting a strong mid‑year surge followed by reduced demand.

**Weekend Shopping vs Channel Preference**

Aim:  
The purpose of this visualization was to explore whether customer shopping behavior differs between weekends and weekdays, specifically in terms of their preference for online versus store purchases.

In [128]:
(3)
chart = px.histogram(
    df,
    x="Is_Weekend",
    color="Order_Type",
    barmode="group",
    color_discrete_sequence=px.colors.qualitative.Pastel,
    title="Weekend Shopping vs Channel Preference"
)

chart.update_traces(marker_line_color="black", marker_line_width=1)
chart.update_layout(xaxis_title="Is_Weekend", yaxis_title="Count")
chart.show()



Findings:  
The grouped count plot shows a slight visual difference between weekday and weekend order counts by channel. However, the correlation analysis (see EDA section) confirms Is_Weekend and Is_Online are essentially uncorrelated (-0.02), so this visual difference shouldn't be read as a strong or reliable pattern in customer behavior.

**Global Market Share Distribution by Country**

Aim:  
The objective of this visualization was to represent the distribution of sales orders across all countries in the dataset, showing each country's individual share of the global market.

In [129]:
(4)
country_counts = df["Country"].value_counts().reset_index()
country_counts.columns = ["Country", "Orders"]

chart = px.pie(
    country_counts,
    names="Country",
    values="Orders",
    title="Global Market Share Distribution by Country",
    color_discrete_sequence=px.colors.qualitative.Pastel1
)

chart.update_traces(
    textinfo="percent+label",
    marker=dict(line=dict(color="black", width=1)),
    showlegend=False
)

chart.show()

Findings:  
The pie chart revealed the relative share of each country in the overall market. The United States showed the highest performance at roughly 45.6%, emerging as the leading contributor to global sales, while Australia, Canada, France, Germany, and the United Kingdom each contributed more modest, fairly even shares

**Customer Age Distribution**

Aim:  
The purpose of this visualization was to analyze the distribution of customers by age, identifying which age ranges contribute most to sales orders.

In [130]:
(5)
age_data = df["Customer_Age"].dropna()

chart = ff.create_distplot(
    [age_data],
    group_labels=["Customer_Age"],
    bin_size=(age_data.max() - age_data.min()) / 20,
    colors=["skyblue"],
    show_rug=False
)

chart.update_layout(
    title="Customer Age Distribution",
    xaxis_title="Customer_Age",
    yaxis_title="Density",
    showlegend=False
)

chart.show()


Findings:  
The chart shows that customer ages are spread out fairly evenly from 18 to 69 — there's no single age group that stands out much more than the others. The density stays roughly the same across the whole range, with just small ups and downs (a bit higher around the mid-20s, a bit lower around the mid-30s and mid-60s).

**Customer Age Spread and Distribution by Country**

Aim:  
The purpose of this visualization was to analyze how customer ages vary across different countries, identifying whether certain regions have younger or older customer bases.

In [131]:
(6)
chart = px.box(
    df,
    x="Country",
    y="Customer_Age",
    color="Country",
    color_discrete_sequence=px.colors.qualitative.Set3,
    title="Customer Age Spread and Distribution by Country"
)

chart.update_layout(
    xaxis_title="Country",
    yaxis_title="Customer Age",
    showlegend=False
)

chart.show()


Findings:  
The boxplot revealed clear differences in age distribution across countries. The United States showed the widest spread and highest concentration of mid‑age customers, making it the top performer in terms of customer engagement. Other countries displayed narrower ranges or lower median ages, highlighting regional variations in customer demographics.

**Buying Behavior Across Gender and Age Groups**

Aim:  
The purpose of this visualization was to analyze how customer purchasing behavior varies across different age groups, segmented by gender.

In [132]:
(7)
chart = px.histogram(
    df,
    x="Age_Group",
    color="Customer_Gender",
    barmode="group",
    color_discrete_sequence=px.colors.diverging.PRGn,
    title="<b>Buying Behavior Across Gender and Age Groups</b>"
)

chart.update_layout(
    xaxis_title="Age Group",
    yaxis_title="Count",
    width=800,
    height=500
)

chart.update_traces(
    marker_line_color="black",
    marker_line_width=1
)

chart.show()


Findings:  
The count plot revealed clear differences in buying behavior. Male customers showed the highest performance in the Young Adults (25–34) group, while female customers displayed stronger participation in the Adults (35–64) group — which is also the single largest segment overall for both genders. This highlights how age and gender together influence sales trends, with younger adult males and mid‑to‑older adult females emerging as the most active contributors.

**Correlation Heatmap**

Aim:  
The purpose of this visualization was to examine the linear relationships among numeric features in the dataset, identifying whether any strong dependencies exist that could influence customer behavior or sales outcomes.

In [133]:
(8)
corr = df.corr(numeric_only=True)

chart = px.imshow(
    corr,
    text_auto=".2f",
    color_continuous_scale="YlGnBu",
    title="<b>Correlation Heatmap</b>",
    labels=dict(color="Correlation")
)

chart.update_layout(width=800, height=600)
chart.update_xaxes(side="bottom")
chart.show()

Findings:  
The heatmap revealed that most variables had very weak or near‑zero correlations. The highest correlation observed was between Customer Age and Is_Weekend (0.02), though it was still negligible. This confirms that the dataset’s numeric features are largely independent, with no strong linear relationships influencing customer behavior.

**Country vs Age Group and Order Type (Subplots)**

Aim:  
The purpose of these subplots was to analyze customer demographics and sales channel preferences across different countries, highlighting regional variations in age groups and order types.

In [134]:
(9)
chart = make_subplots(
    rows=1, cols=2,
    subplot_titles=(
        "<b>Regional Analysis of Age Groups</b>",
        "<b>Cross-Country Variations in Sales Channel Preferences</b>"
    )
)

chart1 = px.histogram(
    df,
    x="Country",
    color="Age_Group",
    barmode="group",
    color_discrete_sequence=px.colors.qualitative.Plotly
)

chart2 = px.histogram(
    df,
    x="Country",
    color="Order_Type",
    barmode="group",
    color_discrete_sequence=px.colors.qualitative.D3
)

for trace in chart1.data:
    chart.add_trace(trace, row=1, col=1)

for trace in chart2.data:
    chart.add_trace(trace, row=1, col=2)

chart.update_layout(
    barmode="group",
    width=1400,
    height=600,
    showlegend=True
)

chart.update_traces(marker_line_color="black", marker_line_width=1)
chart.update_xaxes(tickangle=30)

chart.show()

Findings:

Regional Analysis of Age Groups: The subplot showed that Adults (35–64 years) were the most dominant age group across countries, with the United States leading in overall customer volume.

Cross‑Country Variations in Sales Channel Preferences: The subplot revealed that the United States again showed the highest performance in online orders, while other countries displayed more balanced preferences between online and store channels.

**Multivariate Feature Relationships (Pair Plot)**

Aim:  
The purpose of this visualization was to explore the pairwise relationships among numeric features (Customer Age, Day, Year, Is_Weekend), segmented by order type, to identify any visible patterns or clustering.

In [135]:
(10)
df_numeric = df[["Customer_Age", "Day", "Year", "Is_Weekend"]].copy()
df_numeric["Order_Type"] = df["Order_Type"]

chart = px.scatter_matrix(
    df_numeric,
    dimensions=["Customer_Age", "Day", "Year", "Is_Weekend"],
    color="Order_Type",
    color_discrete_sequence=px.colors.qualitative.Set1,
    title="<b>Multivariate Feature Relationships (Pair Plot)</b>"
)

chart.update_traces(
    diagonal_visible=True,
    opacity=0.6,
    marker=dict(line=dict(width=0.5, color="gray"))
)

chart.update_layout(width=900, height=800)
chart.show()

Findings:  

The scatter matrix reveals that the numeric features do not exhibit strong linear correlations, as most data distributions appear heavily dispersed. However, a distinct structural separation is visible based on the order medium; online orders form highly concentrated clusters across various feature combinations, whereas store orders show a more uniform, scattered distribution. This demonstrates that while the numeric metrics remain largely independent of one another, the mode of purchase serves as the primary differentiating factor within these multivariate relationships.--




**Overall Summary**:  
Our visualizations reveal clear shopping trends: online orders peaked in 2022 before easing back in 2023–2024, while store orders remained fairly stable throughout the period. Seasonal peaks occur mid‑year, and a few countries dominate global sales though others still contribute significantly. Customer age distribution is spread broadly and evenly across all age groups rather than concentrated in any one band, and buying behavior differs across gender and age groups. Correlation analysis shows weekend shopping and online preference are not meaningfully linked (correlation ≈ -0.02), while multivariate plots confirm regional and demographic variations in channel choice.

## **Bike Sales Data Analytics — Capstone Project Summary**
**1**. **Data Analytics Project Summary**

This Capstone Project successfully delivers an automated, code-driven End-to-End Sales Analytics pipeline utilizing Python and Pandas. Over traditional spreadsheet workflows, this pipeline processed **5,263 unique**  **transactional records spanning from January 2020** **to December 2024** to decode global market reach, buyer behaviors, and channel dynamics.

The system systematically eliminated structural text inconsistencies (such as standardizing gender and country typos), resolved missing values across both temporal fields (Day) and demographic fields (Age_Group), and enriched the analytics matrix with high-impact derived features (Is_Weekend, Order_Type, Age_Group, Day/Month/Year) to build a repeatable framework for corporate reporting.

# **2. 5 Key Analytical Insights Explained**
* **Insight 1: The Core Demographical Powerhouse**

Since the raw transactional records are anonymized with no customer names or IDs, Customer_Age serves as the primary demographic variable in this pipeline. The data shows that the **Adults (35–64)** bracket is the strongest contributor to this retail chain, capturing **2,989 orders** out of the 5,263 total records — over half the dataset. Marketing efforts may benefit from prioritizing this age group.

* **Insight 2: Chronological Seasonality & Peak Windows**

A distinct cyclical pattern emerged from the monthly trend analysis. Retail activity does not flow uniformly across the year; instead, it shows a mid-year demand surge that peaks in June before a sharp drop in July. This supports planning inventory and staffing around a Q2 peak window.

* **Insight 3: Year-over-Year Order Stability (2020–2021)**

Order volume remained stable between 2020 and 2021 (1,037 → 1,061 orders, a ~2% shift), showing no significant disruption across this period in the dataset. Since this dataset is synthetic, this stability likely reflects the data generation process rather than a real-world business event — it's included here as a validation check on year-over-year consistency rather than a causal finding.

* **Insight 4: Weekend Activity and Channel Choice**

Cross-referencing weekend activity against order channel shows no meaningful linear relationship — the correlation between Is_Weekend and Is_Online is just **-0.02**, confirming these variables are independent in this dataset. While the grouped histogram shows some visual variation between channels on weekends, this isn't backed by a reliable correlation. Weekend-specific promotions may still be worth A/B testing operationally, but the current data doesn't support them as a proven driver of channel choice.

* **Insight 5: Geographical Concentration and Demographic Spread**

The distribution confirms the **United States commands the largest market footprint, capturing 45.6% (2,402 orders)** of the entire retail pipeline — verified directly against country-level order counts. Regarding customer age, the boxplot analysis (Chart 6) shows some variation in age spread across the 6 countries, with the United States showing the widest range. A precise per-country mean/median comparison would need to be computed directly before claiming demographic uniformity — this is noted as an area for further analysis rather than a confirmed finding.

# **3**. **Overall Insight Summary**
 **1 .Overall Market Condition: Mature & Stabilized – Consistent Operational Baseline**

 Across all analyzed dimensions, the dataset demonstrates a high degree of **market stability and structural consistency** rather than aggressive growth or systemic decline. The strongest empirical evidence of this mature market behavior is visible in the year-over-year transaction volume from 2020 to 2024

 * **2020: 1,037** orders

 * **2021: 1,063** orders

 * **2022: 1,063** orders

 * **2023: 1,063** orders

 * **2024: 1,037** orders

 This near-symmetrical distribution (maintaining a minimal ±2.5% variance, with identical baseline volumes in 2020 and 2024) indicates a **highly predictable and steady-state retail environment**. There is no compounding upward momentum or downward trajectory present over the five-year temporal range, suggesting that the business operates within a well-established, fixed demand cap.

 Rather than macroscopic expansion, the primary operational dynamic identified is an internal channel-mix migration:

* **Online Orders**: Rose from 510 (2020) to a peak of 574 (2022), before normalizing at 523 by 2024.

* **Store Orders**: Moved in an inverse trajectory, balancing the total volume.

 This confirms a consumer reallocation between platforms (Omnichannel Fluctuation) within a static volume constraint, rather than a net increase in market customer acquisition. Furthermore, because the source dataset explicitly tracks order counts without financial metrics (price, revenue, or cost), monetary profitability cannot be evaluated; the pipeline focuses strictly on volume performance.

 **Conclusion**:

 The data reflects a **highly stable and optimized retail infrastructure** characterized by a uniform distribution of order volumes over time. The business exhibits strong structural concentration within a specific demographic core (**Adults aged 35–64**) and a dominant geographic anchor (**United States at 45.6%**), supported by a repeating mid-year seasonal surge that predictably peaks every June. This structural predictability provides a solid baseline for risk-free inventory planning, resource allocation, and targeted demographic optimizations.

# **4. Strategic Business Recommendations & Next Steps**

* **Execute Q2 Inventory Front-Loading**: Supply chain readiness should be increased between March and May to prepare for the June demand peak observed in the monthly trend analysis, tapering afterward given the July drop-off.

* **Test Weekend-Focused Digital Campaigns**: While the current correlation data doesn't show a strong statistical link between weekends and online orders, the visual trend is worth validating through a controlled test (e.g., a time-boxed weekend promotion) rather than assuming it as a confirmed driver.

* **US Prioritize the Core Anchor Market**: With the United States controlling 45.6% of global order volume, expansion and retention budgets should prioritize this market first.

 * **Upgrade Pipeline to Predictive Modeling (Future Scope)**: Extend
 this pipeline with advanced forecasting techniques to estimate monthly transaction volumes, and integrate customer segmentation based on purchase timing, frequency of activity, and overall spending value. This will allow the system to anticipate demand patterns more accurately and track customer retention in a structured way.